pytest.ini

```diff
@@ -15,17 +15,24 @@ filterwarnings=
 	# Ensure ResourceWarnings are emitted
 	default::ResourceWarning
 
+	# Workarounds for pypa/setuptools#3810
+	# Can't use EncodingWarning as it doesn't exist on Python 3.9.
+	# These warnings only appear on Python 3.10+
+	
+	# python/mypy#17057
+	ignore:'encoding' argument not specified::mypy
+	ignore:'encoding' argument not specified::configparser
+	# ^-- ConfigParser is called by mypy,
+	#     but ignoring the warning in `mypy` is not enough
+	#     to make it work on PyPy
+
 	# realpython/pytest-mypy#152
 	ignore:'encoding' argument not specified::pytest_mypy
 
-	# python/cpython#100750
-	ignore:'encoding' argument not specified::platform
-
-	# pypa/build#615
-	ignore:'encoding' argument not specified::build.env
-
-	# dateutil/dateutil#1284
-	ignore:datetime.datetime.utcfromtimestamp:DeprecationWarning:dateutil.tz.tz
+	# Already fixed in pypa/distutils, but present in stdlib
+	ignore:'encoding' argument not specified::distutils
+	# The stdlib test calls distutils.util.byte_compile but the true call location is hidden by pytest.warn context
+	ignore:'encoding' argument not specified::setuptools.tests.test_build_py
```

Comment:
@abravalheri Is this line still needed with the change you did in `setuptools/tests/test_build_py.py` ?
```suggestion
```

Reply:
It should not be needed...

I think it is best to avoid merging the main branch back here...
Instead I have the impression that rebases would work better.

Reply:
If that is OK with you and you don't mind loosing https://github.com/pypa/setuptools/pull/4255/commits/1d439a5835790769c6b674c039623b4c80da594a and https://github.com/pypa/setuptools/pull/4255/commits/1d439a5835790769c6b674c039623b4c80da594a, I can force push, including the fix for the windows tests that I just implemented.